### PR TITLE
Data bucketing, attention scoring and minor fixes

### PR DESCRIPTION
**Pull request by cfrancesco**

### CODE REVIEW

**Review by cschaefer26 — APPROVED** [review]

File: create_training_data.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        phonemized_data.update({file: phonemized_text})
+        
+    print('\nPhonemized metadata samples:')
+    for i in sample_items:
+        print(f'{i}:{phonemized_data[i]}')
+        
+    new_metadata = [''.join([key, '|', phonemized_data[key], '\n']) for key in phonemized_data]
+    shuffled_metadata = np.random.permutation(new_metadata)
+    train_metadata = shuffled_metadata[0:train_len]
+    test_metadata = shuffled_metadata[:test_len]
```
[the remark this answers is not in the document]
Comment (cfrancesco) [reply create_training_data.py:132]: Very odd. Of course

File: config/melgan/data_config.yaml [hunk 2]
```diff
@@ -23,4 +24,4 @@ f_max: 8000
 normalizer: MelGAN                 # which mel normalization to use from utils.audio.py [MelGAN or WaveRNN]
 # TOKENIZER
 phoneme_language: 'en'
```
[the remark this answers is not in the document]
Comment (cfrancesco) [reply config/melgan/data_config.yaml:26]: Changed.

File: preprocessing/text/__init__.py [hunk 3]
```diff
@@ -16,25 +16,25 @@ def __call__(self, input_text):
...
+    def default_pipeline(cls, language, add_start_end, with_stress, njobs=1):
         if language == 'en':
             cleaner = English()
         elif language == 'de':
             cleaner = German()
         else:
             raise ValueError(f'language must be either "en" or "de", not {language}.')
-        phonemizer = Phonemizer(language=language, strip=False, njobs=1, with_stress=with_stress)
+        phonemizer = Phonemizer(language=language, strip=False, njobs=njobs, with_stress=with_stress)
         tokenizer = Tokenizer(sorted(list(_phonemes) + list(_punctuations)), add_start_end=add_start_end)
```
[the remark this answers is not in the document]
Comment (cfrancesco) [reply preprocessing/text/__init__.py:27]: Changed, I add all_phoneme for this purpose precisely.

File: utils/spectrogram_ops.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+
+if __name__ == '__main__':
+    from preprocessing.metadata_readers import get_preprocessor_by_name
+    from preprocessing.datasets import DataReader, AutoregressivePreprocessor, TextMelDataset
+    from preprocessing.text.tokenizer import Tokenizer
+    from preprocessing.text.symbols import all_phonemes
+    from pathlib import Path
+    
+    ljspeech_folder = '/Volumes/data/datasets/LJSpeech-1.1'
```
[the remark this answers is not in the document]
Comment (cfrancesco) [reply utils/spectrogram_ops.py:26]: I might just remove it then, for tests I would write something more principled.

**Pull request merged**